cli/tests/pcluster/delete/test_pcluster_delete.py

```diff
@@ -0,0 +1,141 @@
+    mocker.patch("pcluster.commands.utils.get_cluster_substacks").return_value = stacks
+    mocker.patch("pcluster.commands.utils.get_stack_template").return_value = template
+    mocker.patch("pcluster.commands._persist_cloudwatch_log_group")
+    if template.get("Resources", {}).get("key", {}).get("DeletionPolicy") == "Retain":
+        mocker.patch("pcluster.commands._get_unretained_cw_log_group_resource_keys").return_value = ["key"]
+        expected_persist_cloudwatch_log_groups_call_count = len(stacks)
+    else:
+        mocker.patch("pcluster.commands._get_unretained_cw_log_group_resource_keys").return_value = []
+        expected_persist_cloudwatch_log_groups_call_count = 0
+    commands._persist_cloudwatch_log_groups(FAKE_CLUSTER_NAME)
+    commands.utils.get_cluster_substacks.assert_called_with(FAKE_CLUSTER_NAME)
+    assert_that(commands.utils.get_stack_template.call_count).is_equal_to(len(stacks))
+    assert_that(commands._get_unretained_cw_log_group_resource_keys.call_count).is_equal_to(len(stacks))
+    assert_that(commands._persist_cloudwatch_log_group.call_count).is_equal_to(
+        expected_persist_cloudwatch_log_groups_call_count
+    )
+
+
+@pytest.mark.parametrize(
+    "template",
+    [
+        {},
+        {"Resources": {}},
+        {"Resources": {"key": {}}},
+        {"Resources": {"key": {"DeletionPolicy": "Don't Retain"}}},
+        {"Resources": {"key": {"DeletionPolicy": "Retain"}}},  # Note update_stack_template still called for this
+    ],
+)
+def test_persist_cloudwatch_log_group(mocker, template):
+    """Verify that commands._persist_cloudwatch_log_group behaves as expected."""
+    mocker.patch("pcluster.commands.utils.update_stack_template")
+    stack = {"StackName": FAKE_STACK_NAME, "Parameters": {"Some Key": "Some Values"}}
+
+    if "Resources" not in template:
+        expected_error_message = "Resources"
+    elif "key" not in template.get("Resources"):
+        expected_error_message = "key"
+    else:
+        expected_error_message = None
+
+    if expected_error_message:
+        with pytest.raises(KeyError, match=expected_error_message):
+            commands._persist_cloudwatch_log_group(stack, template, ["key"])
+        assert_that(commands.utils.update_stack_template.called).is_false()
+    else:
+        commands._persist_cloudwatch_log_group(stack, template, ["key"])
+        commands.utils.update_stack_template.assert_called_with(
+            stack.get("StackName"), template, stack.get("Parameters")
+        )
+        assert_that(template["Resources"]["key"]["DeletionPolicy"]).is_equal_to("Retain")
+
+
+@pytest.mark.parametrize(
+    "template,expected_return",
+    [
+        ({}, []),
+        ({"Resources": {}}, []),
+        ({"Resources": {"ResourceOne": {"Type": LOG_GROUP_TYPE, "DeletionPolicy": "Retain"}}}, []),
+        ({"Resources": {"ResourceOne": {"Type": LOG_GROUP_TYPE, "DeletionPolicy": "NotRetain"}}}, ["ResourceOne"]),
```

Comment:
Minor: NotRetain --> Delete ?

Reply:
The purpose was to show that the function doesn't care about the value of the string aside from whether or not it's equal to `Retain`.

Reply:
I added a case that has `Delete` as well.